pkg/scheduler/eventhandlers.go

```diff
@@ -197,6 +197,10 @@ func (sched *Scheduler) deletePodFromSchedulingQueue(obj interface{}) {
 		utilruntime.HandleError(fmt.Errorf("unable to handle object in %T: %T", sched, obj))
 		return
 	}
+
+	// If "pod" is a nominated pod, we should remove it from SchedulingQueue.
+	sched.SchedulingQueue.DeleteNominatedPodIfExists(pod)
+
```

Comment:
I find we delete the pod from `nominatedPods` when we call `sched.SchedulingQueue.Delete`.
https://github.com/kubernetes/kubernetes/blob/c97baa339b5043424522dc7932fc3891f15543cb/pkg/scheduler/internal/queue/scheduling_queue.go#L513-L520

Reply:
Thanks for it! The cluster has this patch, so L202 is not needed. However there is another place that needs to call it. I'll give a detailed analysis for this problem.